### PR TITLE
Feat: api: Api call to get actor cids works for versions < 16

### PR DESCRIPTION
**Pull request by geoff-vball**
## Related Issues
Closes https://github.com/filecoin-project/lotus/issues/8934

## Proposed Changes
`StateActorCodeCIDs` now works for all network/actors versions.
Added a new API call `StateActorManifestCID` to get the manifest cid separately to make `StateActorCodeCIDs` cleaner and more uniform across versions 

## Checklist

Before you mark the PR ready for review, please make sure that:
- [ ] All commits have a clear commit message.
- [ ] The PR title is in the form of of `<PR type>: <area>: <change being made>`
    - example: ` fix: mempool: Introduce a cache for valid signatures`
    - `PR type`: _fix_, _feat_, _INTERFACE BREAKING CHANGE_, _CONSENSUS BREAKING_, _build_, _chore_, _ci_, _docs_,_perf_, _refactor_, _revert_, _style_, _test_
    - `area`: _api_, _chain_, _state_, _vm_, _data transfer_, _market_, _mempool_, _message_, _block production_, _multisig_, _networking_, _paychan_, _proving_, _sealing_, _wallet_, _deps_
- [ ] This PR has tests for new functionality or change in behaviour
- [ ] If new user-facing features are introduced, clear usage guidelines and / or documentation updates should be included in https://lotus.filecoin.io or [Discussion Tutorials.](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] CI is green

### CODE REVIEW

File: chain/actors/actor_cids.go [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+	builtin6 "github.com/filecoin-project/specs-actors/v6/actors/builtin"
+	builtin7 "github.com/filecoin-project/specs-actors/v7/actors/builtin"
+	"github.com/ipfs/go-cid"
+	"golang.org/x/xerrors"
+)
+
+// GetActorCodeID looks up a builtin actor's code CID by actor version and canonical actor name.
+func GetActorCodeID(av Version, name string) (cid.Cid, bool) {
+
+	// Actors V8 and above
```
Comment (arajasek) [inline chain/actors/actor_cids.go:18]: I would explicitly put this in an `if av >= Version8`

File: chain/actors/actor_cids.go [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+		case Version7:
+			return builtin7.VerifiedRegistryActorCodeID, true
+		}
+	}
+
+	return cid.Undef, false
+}
+
+// GetActorCodeIDs looks up a builtin actor's code CID by actor version.
```
Comment (arajasek) [inline chain/actors/actor_cids.go:307]: Comment seems misleading -- it doesn't lookup a specific actor's code CID, but instead returns all builtin actor code CIDs for the given AV?

File: node/impl/full/state.go [hunk 3]
```diff
@@ -1468,25 +1468,26 @@ func (a *StateAPI) StateActorCodeCIDs(ctx context.Context, nv network.Version) (
...
-		actorCID, ok := actors.GetActorCodeID(actorVersion, name)
-		if !ok {
-			return nil, xerrors.Errorf("didn't find actor %v code id for actor version %d", name,
-				actorVersion)
-		}
-		cids[name] = actorCID
+func (a *StateAPI) StateActorManifestCID(ctx context.Context, nv network.Version) (cid.Cid, error) {
+	actorVersion, err := actors.VersionForNetwork(nv)
+	if err != nil {
+		return cid.Undef, xerrors.Errorf("invalid network version")
```
Comment (arajasek) [inline node/impl/full/state.go:1482]: specify the input nv and the error err here

**Review by arajasek — COMMENTED** [review]
Given that we're doing this big method, we should definitely get rid of the individual methods called `GetMultisigActorCodeID` (etc.).

File: chain/actors/actor_cids.go [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+// GetActorCodeID looks up a builtin actor's code CID by actor version and canonical actor name.
+func GetActorCodeID(av Version, name string) (cid.Cid, bool) {
+
+	// Actors V8 and above
+	if av >= Version8 {
+		if cids, ok := GetActorCodeIDsFromManifest(av); ok {
+			c, ok := cids[name]
+			return c, ok
+		}
```
Comment (arajasek) [inline chain/actors/actor_cids.go:23]: Simplify to just `return cids[name]`?

Comment (geoff-vball) [reply chain/actors/actor_cids.go:23]: Can't do that because maps are silly

**Pull request merged**